src/main/java/com/google/devtools/build/lib/analysis/BaseRuleClasses.java

```diff
@@ -329,6 +330,15 @@ public Metadata getMetadata() {
                 .dontCheckConstraints()
                 .nonconfigurable(
                     "special logic for constraints and select: see ConstraintSemantics"))
+        /* <!-- #BLAZE_RULE(toolchain).ATTRIBUTE(target_compatible_with) -->
```

Comment:
Doc updates?

Reply:
Definitely. I was trying to avoid doing too much work before getting a high level buy-in on the architecture.

I'll start fleshing this out some more.

Reply:
At this point I feel pretty confident in the documentation.